x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -1087,38 +1086,51 @@ protected LogicalPlan doRule(LogicalPlan plan) {
                 return plan;
             }
 
-            // Otherwise drop the converted attributes after the alias function, as they are only needed for this function, and
-            // the original version of the attribute should still be seen as unconverted.
-            plan = dropConvertedAttributes(plan, unionFieldAttributes);
+            // In ResolveRefs the aggregates are resolved from the groupings, which might have an unresolved MultiTypeEsField.
+            // Now that we have resolved those, we need to re-resolve the aggregates.
+            if (plan instanceof EsqlAggregate agg) {
```

Comment:
paranoid nit: in 8.15, the aggregates have not yet been consolidated. To be on the safer side, we could have `plan instanceof Aggregate agg` here, in case _somehow_ we create `Aggregate` instead of `EsqlAggregate` in our plans.

Reply:
The same could be said about all usages of EsqlAggregate throughout this class, independent of union-types work. I'd rather keep consistent with the rest of the code. If there is a way for Aggregate to be created that is not EsqlAggregate, it will mess up a lot more than this code.